### DISCUSSION TITLE
libncmdump外部调用直接获取ncm元数据

### DISCUSSION THREAD

**Discussion by WhatDamon** [taurusxin/ncmdump#43] [Ideas]
详见ncmdump的python实现：https://github.com/ww-rm/ncmdump-py
通过相应API在进行dump前获取歌曲元数据信息，建议以json方式传输
包括信息包括但不限于：
- `id` _//歌曲在网易云上的ID_
- `format` _//音频格式，`flac`或者`mp3`_
- `title`
- `artists`
- `album`

还有更多数据，尤其是 `id` 可以用于补全封面或者获取最新的元数据，或者歌词
所有数据随父销毁

**Comment by taurusxin** [comment]
不错的提议，可以提供更多的 API 供调用，最近工作较忙，欢迎 pr。
